backpack/utils/conv_transpose.py

```diff
@@ -33,14 +33,21 @@ def extract_weight_diagonal(module, input, grad_output, N):
 
     input_reshaped = input.reshape(M, -1, spatial_out_numel)
 
+    V_axis, N_axis = 0, 1
     grad_output_viewed = separate_channels_and_pixels(module, grad_output)
     AX = einsum("nkl,vnml->vnkm", (input_reshaped, grad_output_viewed))
-    weight_diagonal = (AX ** 2).sum([0, 1]).transpose(0, 1)
-    weight_diagonal = weight_diagonal.reshape(in_channels, out_channels, *k)
-    return weight_diagonal.transpose(0, 1)
+    N = AX.shape[N_axis]
+    sum_dims = [V_axis, N_axis] if sum_batch else [V_axis]
+    transpose_dims = (V_axis, N_axis) if sum_batch else (V_axis + 1, N_axis + 1)
+    weight_diagonal = (AX ** 2).sum(sum_dims).transpose(*transpose_dims)
+    if sum_batch:
+        weight_diagonal = weight_diagonal.reshape(in_channels, out_channels, *k)
```

Comment:
Can you change that to `view_as(module.weight)`, similar to `extract_weight_diagonal` in `utils/conv.py`?

Reply:
This also does work with `view_as(module.weight)`, I have the error message

Reply:
The intermediate result is transposed, hence the target shape is not `model.shape`